app/src/main/java/com/github/se/cyrcle/ui/map/MapScreen.kt

```diff
@@ -89,7 +106,50 @@ fun MapScreen(
                 mapView.annotations.createPointAnnotationManager(
                     AnnotationConfig(
                         annotationSourceOptions =
-                            AnnotationSourceOptions(clusterOptions = ClusterOptions())))
+                            AnnotationSourceOptions(clusterOptions = ClusterOptions()),
+                        layerId = LAYER_ID))
+
+            annotationManager.addClickListener(
+                OnPointAnnotationClickListener {
+                  removeViewAnnotation = false
+                  viewAnnotationManager.removeAllViewAnnotations()
+
+                  // recenter the camera on the marker if it is not the case already
+                  if (mapViewportState.cameraState?.center != it.point) {
```

Comment:
useless to check if it's not the center, 99% it won't be the case, And if it is already the center, it changes nothing to center it again

Reply:
I agree with this, though I'm not sure it impacts performance much as is.

Reply:
Fixed !